_pytest/terminal.py

```diff
@@ -153,6 +155,16 @@ def __init__(self, config, file=None):
         self._progress_items_reported = 0
         self._show_progress_info = self.config.getini('console_output_style') == 'progress'
 
+    @property
+    def writer(self):
+        warnings.warn(TERMINAL_REPORTER_WARNING, stacklevel=2)
```

Comment:
i propose to rename the internal property and trigger the deprecation also on the changing of self._tw to get bad players in line